src/main/java/com/wcc/platform/repository/postgres/PostgresMentorsRepository.java

```diff
@@ -0,0 +1,166 @@
+
+import com.wcc.platform.domain.cms.attributes.Experience;
+import com.wcc.platform.domain.cms.attributes.Languages;
+import com.wcc.platform.domain.cms.attributes.TechnicalArea;
+import com.wcc.platform.domain.cms.pages.mentorship.Availability;
+import com.wcc.platform.domain.cms.pages.mentorship.MenteeSection;
+import com.wcc.platform.domain.platform.mentorship.Mentor;
+import com.wcc.platform.domain.platform.mentorship.MentorshipType;
+import com.wcc.platform.domain.platform.mentorship.Skills;
+import com.wcc.platform.repository.MentorsRepository;
+import com.wcc.platform.repository.postgres.component.MemberMapper;
+import com.wcc.platform.repository.postgres.component.MentorMapper;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+@Repository
+@RequiredArgsConstructor
+public class PostgresMentorsRepository implements MentorsRepository {
+
+  private static final String SQL_GET_BY_ID = "SELECT * FROM mentors WHERE mentor_id = ?";
+  private static final String SQL_DELETE_BY_ID = "SELECT * FROM mentors WHERE mentor_id = ?";
+  private static final String SQL_GET_BY_EMAIL =
+      "SELECT mentors.* FROM mentors JOIN members ON mentors.mentor_id = members.id WHERE members.email = ?";
+  private static final String SQL_GET_ALL = "SELECT * FROM mentors";
+  private static final String SQL_INSERT =
+      "INSERT INTO mentors (mentor_id, profile_status, bio, years_experience, spoken_languages, is_available) "
+          + "VALUES (?, ?, ?, ?, ?, ?)";
+  private static final String SQL_FIND_ID_BY_EMAIL =
+      "SELECT mentors.mentor_id FROM mentors JOIN members ON mentors.mentor_id = members.id WHERE members.email = ?";
+
+  private static final String MENTOR_ID_COLUMN = "mentor_id";
+  private final JdbcTemplate jdbc;
+  private final MentorMapper mentorMapper;
+  private final PostgresCountryRepository countryRepository;
+  private final MemberMapper memberMapper;
+
+  @Override
+  public Optional<Mentor> findByEmail(String email) {
+    return jdbc.query(
+        SQL_GET_BY_EMAIL,
+        rs -> {
+          if (rs.next()) {
+            return Optional.of(mentorMapper.mapRowToMentor(rs));
+          }
+          return Optional.empty();
+        },
+        email);
+  }
+
+  @Override
+  public List<Mentor> getAll() {
+    return jdbc.query(SQL_GET_ALL, (rs, rowNum) -> mentorMapper.mapRowToMentor(rs));
+  }
+
+  @Override
+  public Long findIdByEmail(String email) {
+    return jdbc.query(
+        SQL_FIND_ID_BY_EMAIL,
+        rs -> {
+          if (rs.next()) {
+            long id = rs.getLong(MENTOR_ID_COLUMN);
+            return Long.valueOf(id);
+          }
+          return null;
+        },
+        email);
+  }
+
+  @Override
+  @Transactional
+  public Mentor create(Mentor entity) {
+    if (entity == null) {
```

Comment:
I this kind of validation should be done in the service, not in the database layer. 

Also you can use Require not null object, so you don't need to add the if.